doctr/datasets/datasets/base.py

```diff
@@ -55,7 +56,13 @@ def __getitem__(self, index: int) -> Tuple[Any, Any]:
             img = self.img_transforms(img)
 
         if self.sample_transforms is not None:
-            img, target = self.sample_transforms(img, target)
+            if isinstance(target, dict):
+                img_transformed = copy_tensor(img)
+                for class_name, bboxes in target.items():
```

Comment:
An OCRDataset comes also with dict as target and has no class names (string labels and bbox annotations) so if we transform samples in this case this should normally fail

Reply:
Yes that's true :fearful:  i'm trying to think how to handle it . open to any suggestions x)

Reply:
As described in the other comments if we handle it like it is done with object_detection (classes comes from model.cfg which is flexible and can be updated with classes from dataset) we could remove this 
Maybe thinking again about this dataset format:
polygons: [....]
classes: ['printed', 'handwritten', 'printed', 'printed', ...]  ? @frgfm @odulcy-mindee @charlesmindee wdyt ?

Reply:
I think we should split the topics of this PR, and discuss this in several issues/discussions to find a proper implementations. There are several breaking change modifications in the PR over quite a lot of files, it's quite hard to ensure the review quality will be both high and exhaustive :sweat_smile:

Reply:
I added another check in addition to if it's a dict to ensure it's the target coming from detection.